src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -262,66 +287,29 @@ CHIP_ERROR OTARequestor::SetupCASESessionManager(chip::FabricIndex fabricIndex)
     return CHIP_NO_ERROR;
 }
 
-// Converted from SendQueryImageCommand()
-void OTARequestor::ConnectToProvider()
+void OTARequestor::ConnectToProvider(OnConnectedAction onConnectedAction)
 {
-    chip::NodeId peerNodeId           = mProviderNodeId;
-    chip::FabricIndex peerFabricIndex = mProviderFabricIndex;
-
-    Server * server           = &(Server::GetInstance());
-    chip::FabricInfo * fabric = server->GetFabricTable().FindFabricWithIndex(peerFabricIndex);
-    if (fabric == nullptr)
-    {
-        ChipLogError(SoftwareUpdate, "Did not find fabric for index %d", peerFabricIndex);
-        return;
-    }
-
-    chip::DeviceProxyInitParams initParams = {
-        .sessionManager = &(server->GetSecureSessionManager()),
-        .exchangeMgr    = &(server->GetExchangeManager()),
-        .idAllocator    = &(server->GetSessionIDAllocator()),
-        .fabricInfo     = fabric,
-        // TODO: Determine where this should be instantiated
-        .imDelegate = chip::Platform::New<chip::Controller::DeviceControllerInteractionModelDelegate>(),
-    };
-
-    chip::OperationalDeviceProxy * operationalDeviceProxy =
-        chip::Platform::New<chip::OperationalDeviceProxy>(initParams, fabric->GetPeerIdForNode(peerNodeId));
-    if (operationalDeviceProxy == nullptr)
-    {
-        ChipLogError(SoftwareUpdate, "Failed in creating an instance of OperationalDeviceProxy");
-        return;
-    }
-
-    server->SetOperationalDeviceProxy(operationalDeviceProxy);
+    CHIP_ERROR err = SetupCASESessionManager(mProviderFabricIndex);
+    VerifyOrReturn(err == CHIP_NO_ERROR,
+                   ChipLogError(SoftwareUpdate, "Cannot setup CASESessionManager: %" CHIP_ERROR_FORMAT, err.Format()));
```

Comment:
Same comment as before. API surface of OTARequestor should allow clients to capture their metrics re: failures